x/evmutil/keeper/bank_keeper.go

```diff
@@ -0,0 +1,238 @@
+// of the 18 decimal akava coin on the evm.
+// x/evm consumes gas and send coins by minting and burning akava coins in its module
+// account and then sending the funds to the target account.
+// This keeper uses both the ukava coin and a separate akava balance to manage the
+// extra percision needed by the evm.
+type EvmBankKeeper struct {
+	akavaKeeper Keeper
+	bk          types.BankKeeper
+	ak          types.AccountKeeper
+}
+
+func NewEvmBankKeeper(akavaKeeper Keeper, bk types.BankKeeper, ak types.AccountKeeper) EvmBankKeeper {
+	return EvmBankKeeper{
+		akavaKeeper: akavaKeeper,
+		bk:          bk,
+		ak:          ak,
+	}
+}
+
+// GetBalance returns the total **spendable** balance of akava for a given account by address.
+func (k EvmBankKeeper) GetBalance(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Coin {
+	if denom != EvmDenom {
+		panic(fmt.Errorf("only evm denom %s is supported by EvmBankKeeper", EvmDenom))
+	}
+
+	spendableCoins := k.bk.SpendableCoins(ctx, addr)
+	ukava := spendableCoins.AmountOf(CosmosDenom)
+	akava := k.akavaKeeper.GetBalance(ctx, addr)
+	total := ukava.Mul(ConversionMultiplier).Add(akava)
+	return sdk.NewCoin(EvmDenom, total)
+}
+
+// SendCoinsFromModuleToAccount transfers akava coins from a ModuleAccount to an AccAddress.
+// It will panic if the module account does not exist. An error is returned if the recipient
+// address is black-listed or if sending the tokens fails.
+func (k EvmBankKeeper) SendCoinsFromModuleToAccount(ctx sdk.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins) error {
+	ukava, akava, err := SplitAkavaCoins(amt)
+	if err != nil {
+		return err
+	}
+
+	if ukava.Amount.IsPositive() {
+		if err := k.bk.SendCoinsFromModuleToAccount(ctx, senderModule, recipientAddr, sdk.NewCoins(ukava)); err != nil {
+			return err
+		}
+	}
+
+	if err := k.akavaKeeper.SendBalance(ctx, k.GetModuleAddress(senderModule), recipientAddr, akava); err != nil {
+		return err
+	}
+
+	return k.ConvertAkavaToUkava(ctx, recipientAddr)
+}
+
+// SendCoinsFromAccountToModule transfers akava coins from an AccAddress to a ModuleAccount.
+// It will panic if the module account does not exist.
+func (k EvmBankKeeper) SendCoinsFromAccountToModule(ctx sdk.Context, senderAddr sdk.AccAddress, recipientModule string, amt sdk.Coins) error {
+	ukava, akavaNeeded, err := SplitAkavaCoins(amt)
+	if err != nil {
+		return err
+	}
+
+	// convert 1 ukava to akava if sender account does not have enough akava to cover
+	akavaBal := k.akavaKeeper.GetBalance(ctx, senderAddr)
+	if akavaBal.LT(akavaNeeded) {
+		if err := k.ConvertOneUkavaToAkava(ctx, senderAddr); err != nil {
+			return err
+		}
+	}
+
+	if ukava.IsPositive() {
+		if err := k.bk.SendCoinsFromAccountToModule(ctx, senderAddr, recipientModule, sdk.NewCoins(ukava)); err != nil {
+			return err
+		}
+	}
+
+	return k.akavaKeeper.SendBalance(ctx, senderAddr, k.GetModuleAddress(recipientModule), akavaNeeded)
+}
```

Comment:
Do these methods both need the `ConvertAkavaToUkava` and `ConvertOneUkavaToAkava` sections?
Otherwise the module account could end up with more than 1ukava worth of akava?

Reply:
the module should always be a passthrough account so no need to have `ConvertAkavaToUkava` here when sending funds to it. ethermint will burn the sent funds right after sending it to the module.

Reply:
Same with send from module to account, there should always be enough because ethermint mints the coins onto the module first.